### PR TITLE
Np 45284 person nvi data

### PR DESCRIPTION
**Pull request by torarnet**

### CODE REVIEW

**Comment by torarnet** [comment]
Feature getting large so i will continue in new branch. Current functionality is model classes and GET Person NVI data. The unit test and json shows what i have done

File: cristin-person/src/main/java/no/unit/nva/cristin/person/model/cristin/CristinNviInstitutionUnit.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import static no.unit.nva.cristin.model.JsonPropertyNames.INSTITUTION;
+import static no.unit.nva.cristin.model.JsonPropertyNames.UNIT;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Optional;
+import java.util.Set;
+import no.unit.nva.cristin.model.CristinUnit;
+import no.unit.nva.model.Organization;
+
+public record CristinNviInstitutionUnit(@JsonProperty(INSTITUTION) CristinPersonNviInstitution institution,
```
Comment (StigNorland) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/model/cristin/CristinNviInstitutionUnit.java:11]: Records serialize to json without any markups

File: cristin-person/src/main/java/no/unit/nva/cristin/person/model/cristin/CristinPersonNvi.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.time.Instant;
+import java.util.Optional;
+import no.unit.nva.commons.json.JsonSerializable;
+import no.unit.nva.cristin.person.model.nva.PersonNvi;
+
+public record CristinPersonNvi(@JsonProperty(VERIFIED_BY) CristinPersonSummary verifiedBy,
+                               @JsonProperty(VERIFIED_AT) CristinNviInstitutionUnit verifiedAt,
+                               @JsonProperty(VERIFIED_DATE) Instant verifiedDate)
+    implements JsonSerializable {
```
Comment (StigNorland) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/model/cristin/CristinPersonNvi.java:12]: Intellij crashed when I tried `JsonSerializable` on a record
I will try again...

File: cristin-person/src/main/java/no/unit/nva/cristin/person/model/cristin/CristinPersonSummary.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import static no.unit.nva.cristin.model.Constants.BASE_PATH;
+import static no.unit.nva.cristin.model.Constants.DOMAIN_NAME;
+import static no.unit.nva.cristin.model.Constants.HTTPS;
+import static no.unit.nva.cristin.model.Constants.PERSON_PATH_NVA;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.net.URI;
+import no.unit.nva.cristin.person.model.nva.PersonSummary;
+import nva.commons.core.paths.UriWrapper;
+
+public record CristinPersonSummary(@JsonProperty(FIRST_NAME) String firstName,
```
Comment (StigNorland) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/model/cristin/CristinPersonSummary.java:13]: I really prefer this way of declaring a record, one line for each field, just like we declare a normal class.
It is much easier to read.

File: cristin-person/src/test/java/no/unit/nva/cristin/person/fetch/FetchCristinPersonHandlerTest.java [hunk 4]
```diff
@@ -306,6 +311,22 @@ void shouldNotHaveEmploymentFieldInResponseWhenNotInUpstreamPayload() throws IOE
         assertThat(actual.getEmployments(), equalTo(null));
     }
 
+    @Test
+    void shouldReturnResponseContainingNviDataWhenPresentInUpstream() throws Exception {
```
Comment (StigNorland) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/fetch/FetchCristinPersonHandlerTest.java:315]: Always something new to learn from your tests....

**Review by StigNorland — APPROVED** [review]
I would prefere that all records was formatted the same wat, preferable with one field on each line, easier to read.
and you can also consider removing exxess json markup, but that is up to you

**Review by anetteOlli — APPROVED** [review]

File: cristin-person/src/main/java/no/unit/nva/cristin/person/model/cristin/CristinPersonNvi.java (hunk 2 above)
Comment (torarnet) [reply cristin-person/src/main/java/no/unit/nva/cristin/person/model/cristin/CristinPersonNvi.java:12]: Works on my machine:

````
@Test
void testtest() {
    var json = new CristinPersonNvi(
        new CristinPersonSummary(randomString(), randomString(), null, "1234"),
        new CristinNviInstitutionUnit(null, CristinUnit.fromCristinUnitIdentifier("183.5.5.5")),
        null
    );

    System.out.println(json.toJsonString());
}
````

File: cristin-person/src/test/java/no/unit/nva/cristin/person/fetch/FetchCristinPersonHandlerTest.java (hunk 4 above)
Comment (torarnet) [reply cristin-person/src/test/java/no/unit/nva/cristin/person/fetch/FetchCristinPersonHandlerTest.java:315]: What did you learn?

**Pull request merged**